src/sdk/pynni/nni/trial.py

```diff
@@ -107,6 +108,13 @@ def report_intermediate_result(metric):
         serializable object.
     """
     global _intermediate_seq
```

Comment:
Hi @liuzhe-lz  https://github.com/microsoft/nni/pull/3022#issuecomment-716040368
> Ooops, seems I missed the description.
> 
> For the integration test, I suppose the test case only requires a raw kubernetes cluster without additional setup steps? If that's the case I think we can deploy the pipeline on NNI's cluster. Please provide a pipeline script like [this one](https://github.com/microsoft/nni/blob/master/test/pipelines/pipelines-it-kubeflow.yml), and I will try it on `dev-adl` branch.

Yup we can provide that kinda pipeline. 

A question: I don't see it provides `kubeconfig` for authorization to a Kubernetes cluster. Do you know where it is?

> For intermediate sequence, how should user create or get the accumulator object? Or if it's simply a integral number? I don't see it in the example. If it's a simple number maybe we can set it via environment variable, and only modify line 99 in `trial.py`.

From NNI's perspective, we can rephrase this problem in this general way (ignoring any AdaptDL logics):

* A user needs from NNI side a public interface (like `set_intermediate_seq(seq_id: int)`) in order to set the value of this `global _intermediate_seq`, for all other sequence-changing semantics that is not expected by NNI. 

How do you think of this approach? :smile:

Reply:
Done :) https://github.com/microsoft/nni/blob/81d6c2917466bcacfcb0e51fe6b156ad32b8f26d/src/sdk/pynni/nni/trial.py#L102